packages/core/src/list-control/ListControlState.ts

```diff
@@ -51,6 +51,10 @@ export type ListControlProps<Item> = {
    * Callback used to convert an option's `value` to a string. This is needed when the value is different to the display value or the value is not a string.
    */
   valueToString?: (item: Item) => string;
+  /**
+   * If true, options will be selected on tab key press.
+   */
+  selectOnTab?: boolean;
```

Comment:
I don't think this is the correct place to add the type? Right now we only have the logic in ComboBox but this will a prop to Dropdown too.

![image](https://github.com/jpmorganchase/salt-ds/assets/12938082/d0b3ee47-8e98-4d90-a4f7-3f6d0f40313c)